mlflow/openai/__init__.py

```diff
@@ -437,37 +504,44 @@ def __init__(self, model):
             )
         self.model = model
 
+    def format_messages(self, messages, params_list):
+        messages = [_Message(m) for m in messages]
+        return [[m.format(**params) for m in messages] for params in params_list]
+
+    def get_params_list(self, data, variables):
+        if len(variables) == 1:
+            variable = variables[0]
+            if variable in data.columns:
+                return data[[variable]].to_dict(orient="records")
+            else:
+                iter_string_columns = (c for c, v in data.iloc[0].items() if isinstance(v, str))
+                first_string_column = next(iter_string_columns)
+                return [{variable: s} for s in data[first_string_column]]
+        else:
+            return data[variables].to_dict(orient="records")
+
     def predict(self, data):
-        import pandas as pd
         from mlflow.openai.api_request_parallel_processor import process_api_requests
 
-        if isinstance(data, pd.DataFrame):
-            if "content" not in data or "role" not in data:
-                raise mlflow.MlflowException.invalid_parameter_value(
-                    "Input dataframe must contain columns 'content' and 'role'",
-                )
-            messages = data.to_dict(orient="records")
-        elif isinstance(data, list) and all(isinstance(d, dict) for d in data):
-            if not all(map(_has_content_and_role, data)):
-                raise mlflow.MlflowException.invalid_parameter_value(
-                    "Input list of dictionaries must contain keys 'content' and 'role'",
-                )
-            messages = data
+        messages = self.model.get("messages", [])
+        variables = _parse_variables(messages)
```

Comment:
Do we have to do this parsing every time `predict()` is called, or can we parse once when the wrapper is constructed?

Reply:
Great suggestion! 

> can we parse once when the wrapper is constructed?

yes, made this change :)